src/annotator/integrations/pdf.js

```diff
@@ -309,14 +334,78 @@ export class PDFIntegration {
   }
 
   /**
+   * Scroll to the location of an anchor in the PDF.
+   *
+   * If the anchor refers to a location that is an un-rendered page far from
+   * the viewport, then scrolling happens in three phases. First the document
+   * scrolls to the approximate location indicated by the placeholder anchor,
+   * then `scrollToAnchor` waits until the page's text layer is rendered and
+   * the annotation is re-anchored in the fully rendered page. Then it scrolls
+   * again to the final location.
+   *
    * @param {Anchor} anchor
    */
-  scrollToAnchor(anchor) {
-    // TODO - Account for pages changing state between un-rendered and rendered
-    // while scrolling.
-    const highlights = /** @type {Element[]} */ (anchor.highlights);
-    return new Promise(resolve => {
-      scrollIntoView(highlights[0], resolve);
-    });
+  async scrollToAnchor(anchor) {
+    const annotation = anchor.annotation;
+    const isPlaceholder = anchorIsPlaceholder(anchor);
+    const offset = this._anchorOffset(anchor);
+    if (offset === null) {
+      return;
+    }
+
+    await scrollElement(this.contentContainer(), offset);
+
+    if (isPlaceholder) {
+      const anchor = await this._waitForAnnotationToBeAnchored(
+        annotation,
+        this._reanchoringWait
+      );
+      if (!anchor) {
+        return;
+      }
+      const offset = this._anchorOffset(anchor);
```

Comment:
Is there any value in making sure that this `offset` is different from the estimated/placeholder offset (and not re-scrolling if they are equivalent), or is that so unlikely as to be not worth the check?

Reply:
The scroll delay is proportional to the difference between current and final offset, so if the two are identical it will simply be instant.